ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -230,15 +230,32 @@ func (scbuilder *SiteConfigBuilder) instantiateCR(target string, originalTemplat
 
 	// Sanity-check the resulting metadata to ensure it's valid compared to the non-overridden original CR:
 	originalMetadata := originalCR["metadata"].(map[string]interface{})
+
+	// Sanity-check the overridden metadata object to ensure that it's instantiated correctly
+	if overriddenCR["metadata"] == nil {
+		return override, fmt.Errorf("Overriden template metadata in %q is not specified!", overridePath)
+	}
```

Comment:
Another way to do this same check is to use the more lenient type assertion syntax like this:

```
overriddenMetadata, ok := overriddenCR["metadata"].(map[string]interface{})
if !ok {
  return override, err
}
```
(https://go.dev/tour/methods/15)

In fact, you probably don't even have to check `if !ok` since in that more lenient check, it actually returns an empty map without panicking, so you can just change line 238 to:

```
overriddenMetadata, _ := overriddenCR["metadata"].(map[string]interface{})
```
and it may just avoid panic (and hit the pre-existing error case just below) if there's a mismatch...